actors/user/src/main/java/org/sunbird/user/actors/UserManagementActor.java

```diff
@@ -776,4 +780,56 @@ private static void handleGetFrameworkDetails(String frameworkId) {
       }
     }
   }
+
+  private void throwRecoveryParamsMatchException(String type, String recoveryType){
+      ProjectLogger.log(String.format("UserManagementActor:throwParamMatchException:%s should not same as primary %s",recoveryType,type),LoggerEnum.ERROR.name());
+      ProjectCommonException.throwClientErrorException(
+              ResponseCode.recoveryParamsMatchException,
+              MessageFormat.format(ResponseCode.recoveryParamsMatchException.getErrorMessage(),recoveryType,type));
+  }
+
+  private void validateRecoveryEmailPhone(Map<String,Object>userDbRecord,Map<String,Object>userReqMap){
+    String userPrimaryPhone=(String)userDbRecord.get(JsonKey.PHONE);
+    String userPrimaryEmail=(String) userDbRecord.get(JsonKey.EMAIL);
+    String recoveryEmail=(String)userReqMap.get(JsonKey.RECOVERY_EMAIL);
+    String recoveryPhone=(String)userReqMap.get(JsonKey.RECOVERY_PHONE);
+    if(userReqMap.containsKey(JsonKey.RECOVERY_EMAIL) && Matcher.matchIdentifiers(userPrimaryEmail,recoveryEmail)){
```

Comment:
instead of  userReqMap.containsKey(JsonKey.RECOVERY_EMAIL) you can used  "recoveryEmail "  noy null and empty